app/uk/gov/hmrc/hec/repos/HECTaxCheckStore.scala

```diff
@@ -100,6 +107,45 @@ class HECTaxCheckStoreImpl @Inject() (
       }
     )
 
+  /**
+    * Fetch existing tax check codes for the specified GGCredId
+    * @param ggCredId The government gateway ID
+    * @param hc header information
+    * @return A list of tax check code details
+    */
+  def getTaxCheckCodes(ggCredId: GGCredId)(implicit
+    hc: HeaderCarrier
+  ): EitherT[Future, Error, List[HECTaxCheck]] =
+    EitherT(
+      preservingMdc {
+        cacheRepository
+          .find(
+            s"data.$key.taxCheckData.applicantDetails.ggCredId" -> ggCredId.value
```

Comment:
does this make sense @AndyHWChung? or is there a nicer way to do this?

Reply:
I think that'd work - another option would be to stick gg cred id in the document at a higher level but that may not buy us too much over what you've already got here

Reply:
where exactly do we do this?

Reply:
if you're overriding that `def` then it could just be anywhere in the class - it'll get picked up somewhere at the startup of the class

Reply:
sorry, not sure I follow, where is the `ensureIndexes` method defined?

Reply:
oh whoops, it's not part of `HECTaxCheckStore` - it's part of the underlying `CacheMongoRepository`. And actually, it looks like it's just `indexes` you want to override. So, 
```scala
new CacheMongoRepository("hecTaxChecks", expireAfter.toSeconds)(
      mongo.mongoConnector.db,
      ec
    )
```
becomes
```scala
new CacheMongoRepository("hecTaxChecks", expireAfter.toSeconds)(
      mongo.mongoConnector.db,
      ec
    ) {
      override def indexes: Seq[Index] = things
    }
```

Reply:
thanks, I should have looked deeper 🤦‍♀️

Reply:
I think that was mostly me telling you the wrong info :laughing: sorry!

Reply:
index added - https://github.com/hmrc/hec/pull/18/commits/28dd50f5668d3f6fe1911f0eae564cc7be631abb